lxd/resource_lxd_container.go

```diff
@@ -247,13 +255,24 @@ func resourceLxdContainerCreate(d *schema.ResourceData, meta interface{}) error
 	}
 
 	// build API request
-	createReq := api.ContainersPost{}
+	createReq := api.InstancesPost{}
 	createReq.Name = name
 	createReq.Profiles = profiles
 	createReq.Config = config
 	createReq.Devices = devices
 	createReq.Ephemeral = ephem
 
+	instanceType := d.Get("type").(string)
+	if instanceType == "container" {
+		createReq.Type = api.InstanceTypeContainer
+	} else if instanceType == "virtual-machine" {
+		createReq.Type = api.InstanceTypeVM
+	} else if instanceType == "" {
+		createReq.Type = api.InstanceTypeAny
+	} else {
+		return fmt.Errorf("invalid type: %s", instanceType)
+	}
```

Comment:
I think an `else if` clause for `any` needs to be added?

Reply:
Thanks for the hint. I forgot to remove `any` at all, because LXD will always change `any` to either `container` or `virtual-machine`. This confuses terraform as it will always result in new changed to apply. The default blank string behaves like `any`.

Reply:
Do you think having a blank option is intuitive to users? Meaning, if a user chooses a blank string (or not set the option at all), the outcome will be what the user expects?

Reply:
A blank string as the attribute value is invalid as it is rejected by `resourceLxdValidateInstanceType`:

```
Error: Instance must have a type of: [container virtual-machine]

  on main.tf line 15, in resource "lxd_container" "testtest":
  15: resource "lxd_container" "testtest" {
            type = ""
```

If basically means, if `type` is given, use it, otherwise give me any kind of instance. I am not sure if LXD has a setting for a default instance type, but that would be supported here without change.